src/opencl_common.c

```diff
@@ -567,44 +581,51 @@ static int start_opencl_device(int sequential_id, int *err_type)
 
 	max_group_size = get_device_max_lws(sequential_id);
 
-	// Get the platform properties
-	properties[0] = CL_CONTEXT_PLATFORM;
-	properties[1] = (cl_context_properties)
-	                platforms[get_platform_id(sequential_id)].platform;
-	properties[2] = 0;
+	do {
+		// Get the platform properties
+		properties[0] = CL_CONTEXT_PLATFORM;
+		properties[1] = (cl_context_properties)
+			platforms[get_platform_id(sequential_id)].platform;
+		properties[2] = 0;
+
+		// Setup context and queue
+		context[sequential_id] = clCreateContext(properties, 1,
+			&devices[sequential_id], NULL, NULL, &ret_code);
+
+		if (ret_code != CL_SUCCESS) {
+			fprintf(stderr, "%u: Error creating context for device %d "
+			        "(%d:%d): %s, %s\n",
+			        NODE, sequential_id,
+			        get_platform_id(sequential_id),
+			        get_device_id(sequential_id), get_error_name(ret_code),
+			        retry < 10 ? "retrying" : "giving up");
+			if (++retry >= 10)
+				error();
+			usleep(NODE * 100);
+		}
+	} while (ret_code != CL_SUCCESS);
+
+	retry = 0;
+	do {
+		queue[sequential_id] = clCreateCommandQueue(context[sequential_id],
+		                       devices[sequential_id], 0, &ret_code);
+
+		if (ret_code != CL_SUCCESS) {
+			fprintf(stderr, "%u: Error creating command queue for "
+			        "device %d (%d:%d): %s, %s\n", NODE,
+			        sequential_id, get_platform_id(sequential_id),
+			        get_device_id(sequential_id), get_error_name(ret_code),
+			        retry < 10 ? "retrying" : "giving up");
+			if (++retry >= 10)
+				error();
+			usleep(NODE * 100);
+		}
+	} while (ret_code != CL_SUCCESS);
```

Comment:
See [1]

Reply:
This is not needed for the nvidia/MPI race I've seen but was included in some early version of my workarounds for that. I just threw it in there in case it helps this fork issue.